### PR TITLE
fix: CourseAccessRole is always case-insensitive

### PR DESCRIPTION
**Pull request by shadinaif**
* fix: key and collation of `CourseAccessRole` in tests: SQLite is performing `in` operations as case-sensitive unless it is explicitly defined as `NOCASE`. We need to test scenarios where `org` in roles is not the same as the one in organizations (case-sensitivity-wise)
* fix: test-data to include case-insensitive orgs: this will pinpoint python code that compares orgs as case-sensitive
* fix: `CourseAccessRole` is always case-insensitive

### CODE REVIEW

File: futurex_openedx_extensions/dashboard/statistics/courses.py [hunk 1]
```diff
@@ -32,9 +32,9 @@ def get_courses_count(
         fx_permission_info, visible_filter=visible_filter, active_filter=active_filter
     )
 
-    return q_set.values('org').annotate(
+    return q_set.values(org_upper_case=Upper('org')).annotate(
         courses_count=Count('id')
-    ).order_by('org')
+    ).order_by(Upper('org'))
```
Comment (OmarIthawi) [inline futurex_openedx_extensions/dashboard/statistics/courses.py:37]: To match domain names, what do you think?

```suggestion
    ).order_by(Lower('org'))
```

File: futurex_openedx_extensions/dashboard/statistics/learners.py [hunk 2]
```diff
@@ -195,7 +196,7 @@ def get_learners_count(fx_permission_info: dict) -> Dict[int, Dict[str, int]]:
     }
     for tenant_id in fx_permission_info['permitted_tenant_ids']:
         result[tenant_id]['learners_count_per_org'] = {
-            item['org']: item['learners_count']
+            item['org_upper_case'].upper(): item['learners_count']
```
Comment (OmarIthawi) [inline futurex_openedx_extensions/dashboard/statistics/learners.py:199]:
```suggestion
            item['org_lower_case'].lower(): item['learners_count']
```

**Review by OmarIthawi — CHANGES_REQUESTED** [review]
Thanks @shadinaif! Sounds like a big feat!

If it's not troubling at all, would you mind changing it to lowercase because that's the preferred case for organizations and course IDs. Domain names are lower case and it's very useful.

File: futurex_openedx_extensions/dashboard/statistics/courses.py (hunk 1 above)
Comment (shadinaif) [reply futurex_openedx_extensions/dashboard/statistics/courses.py]: done

File: futurex_openedx_extensions/dashboard/statistics/learners.py (hunk 2 above)
Comment (shadinaif) [reply futurex_openedx_extensions/dashboard/statistics/learners.py]: done, everywhere. Even in tests, to keep consistency and clear readability. But I left some test-data in upper-case which helps to detect failing code. Nothing is failing 🥳

**Review by OmarIthawi — APPROVED** [review]
Thank you!! Looks great!!

**Pull request merged**